packages/components/src/color-palette/index.js

```diff
@@ -125,6 +127,56 @@ export function CustomColorPickerDropdown( { isRenderedInSidebar, ...props } ) {
 	);
 }
 
+const extractColorNameFromCurrentValue = (
+	currentValue,
+	colors = [],
+	showMultiplePalettes = false
+) => {
+	if ( ! currentValue ) {
+		return '';
+	}
+
+	let colorNameToReturn;
+
+	if ( showMultiplePalettes ) {
+		for ( const {
+			name: colorPaletteName,
+			colors: colorPaletteColors,
+		} of colors ) {
+			for ( const {
+				name: colorName,
+				color: colorValue,
+			} of colorPaletteColors ) {
+				if (
+					colord( currentValue ).toHex() ===
+					colord( colorValue ).toHex()
+				) {
+					return sprintf(
+						// translators: %1$s: The name of the color e.g: "vivid red"; %2$s: the color palette name (e.g. 'Theme', 'Default', 'Custom'...).
+						__( '%1$s (%2$s)' ),
+						colorName,
+						colorPaletteName
+					);
```

Comment:
Not sure what to when multiple palettes are passed to the `ColorPalette` component.

For now, i've decided to show the `colorPaletteName` (i.e. `Theme`, `Default`, `Custom`) in between parenthesis after the color name. This was mainly triggered by the fact that `Theme` colors have names like "primary", "secondary", "background"...

Other options could be:

- do not display the `colorPaletteName`
- ignore colors coming from the `Theme` palette when trying to match them with a name (although, are we guaranteed that the theme palette is always going to be called `Theme`? Will it change in other locales?)